src/workspace-data/data-table/shared/UriViewerLink.ts

```diff
@@ -26,7 +26,7 @@ export const UriViewerLink = (props: UriViewerLinkProps): ReactNode => {
           setModalOpen(true);
         },
       },
-      [isGsUri(uri) || isAzureUri(uri) ? _.last(uri.split(/\/\b/)) : uri]
+      [typeof uri === 'string' && (isGsUri(uri) || isAzureUri(uri)) ? _.last(uri.split(/\/\b/)) : uri]
```

Comment:
Checking the `typeof` here is preferred over using `uri.toString().split…`. When using `uri.toString().split`, the UI renders incorrect parts of the nested arrays. I hope to be able to remove this clause in the longer-term fix.